index.bs

```diff
@@ -558,8 +558,13 @@ Instances of <code>ReadableStream</code> are created with the internal slots des
 </div>
 
 <ol>
-  <li> If <b>this</b>@\[[reader]] is not <b>undefined</b>, return <b>this</b>@\[[reader]]@\[[lockReleased]].
-  <li> Return <b>this</b>@\[[readyPromise]].
+  <li> If <b>this</b>@\[[readableStreamReader]] is not <b>undefined</b>, return the result of transforming
+    <b>this</b>@\[[readableStreamReader]]@\[[lockReleased]] by a fulfillment handler that returns <b>this</b>@\[[readyPromise]].
+  <li> Return the result of transforming <b>this</b>@\[[readyPromise]] by the following fulfillment handler:
+    <ol>
+      <li> If <b>this</b>@\[[readableStreamReader]] is <b>undefined</b>, return <b>undefined</b>.
+      <li> Otherwise, return <b>this</b>@\[[readableStreamReader]]@\[[lockReleased]].
+    </ol>
```

Comment:
I think this is the base case, and so should directly use [[readyPromise]]?

Regardless, I'll be able to use your scenario to write a test case and prove one way or the other.